src/nvs.rs

```diff
@@ -352,6 +352,305 @@ impl<T: NvsPartitionId> EspNvs<T> {
 
         Ok(true)
     }
+
+    pub fn get_blob<'a>(
+        &self,
+        name: &str,
+        buf: &'a mut [u8],
+    ) -> Result<Option<&'a [u8]>, EspError> {
+        let c_key = CString::new(name).unwrap();
+
+        // check for blob value, by getting blob length
+        let mut len = 0;
+        match unsafe { nvs_get_blob(self.1, c_key.as_ptr(), ptr::null_mut(), &mut len as *mut _) } {
+            ESP_ERR_NVS_NOT_FOUND => Ok(None),
+            err => {
+                // bail on error
+                esp!(err)?;
+
+                len = buf.len();
+
+                // fetch value if no error
+                esp!(unsafe {
+                    nvs_get_blob(
+                        self.1,
+                        c_key.as_ptr(),
+                        buf.as_mut_ptr() as *mut _,
+                        &mut len as *mut _,
+                    )
+                })?;
+
+                Ok(Some(&buf[..len]))
+            }
+        }
+    }
+
+    pub fn set_blob(&mut self, name: &str, buf: &[u8]) -> Result<(), EspError> {
+        let c_key = CString::new(name).unwrap();
+
+        // start by just clearing this key
+        unsafe { nvs_erase_key(self.1, c_key.as_ptr()) };
+
+        esp!(unsafe { nvs_set_blob(self.1, c_key.as_ptr(), buf.as_ptr().cast(), buf.len()) })?;
+
+        esp!(unsafe { nvs_commit(self.1) })?;
+
+        Ok(())
+    }
+
+    pub fn str_len(&self, name: &str) -> Result<Option<usize>, EspError> {
+        let c_key = CString::new(name).unwrap();
+
+        #[allow(unused_assignments)]
+        let mut len = 0;
+
+        match unsafe { nvs_get_str(self.1, c_key.as_ptr(), ptr::null_mut(), &mut len as *mut _) } {
+            ESP_ERR_NVS_NOT_FOUND => Ok(None),
+            err => {
+                // bail on error
+                esp!(err)?;
+
+                Ok(Some(len))
+            }
+        }
+    }
+
+    pub fn get_str<'a>(&self, name: &str, buf: &'a mut [u8]) -> Result<Option<&'a str>, EspError> {
+        let c_key = CString::new(name).unwrap();
+
+        #[allow(unused_assignments)]
+        let mut len = 0;
+        match unsafe {
+            len = buf.len();
```

Comment:
Nit: merge `len = buf.len()` with `let mut len = 0;`. There is no reason why the former should be in the `unsafe` block, and separate from the initializer. This will allow you to get rid of the `unused-assigments` warning, which is more or less telling you exactly that.